### DISCUSSION TITLE
PlotSegment X Axis Sorting Issue

### DISCUSSION THREAD

**Discussion by yilmazcihann** [zBritva/charticulator-visual-community#33] [General]
Hello,

I am using Charticulator by Ilfat Galiev in Power BI via Power BI App Source visual. I am trying build a PlotSegment and I select a column data for X Axis. The column name for PlotSegment X Axis Data is BrandName (from DimBrand table) but I want to sort this column by SortOrder column (from DimBrand table) which in same table. SortOrder column is Int64 in SSAS Tabular Model which i connect with live connection. 

As i see, Plot Segment > X Axis > Order By action is sorting this column like 1, 10, 11, 2, 21, 22, 3, 4, 5, 6, 7, 8, 9... 
I expect like 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 21, 22 ...
Is this a known issue or a bug? I also add an image for this issue. 

I am waiting for your quick support.

Thanks in advance.
Cihan Yılmaz

![charticulator-x-axis-order-by-issue](https://github.com/user-attachments/assets/e7966b99-d802-4a79-ac16-6a9737144f7d)

**Comment by zBritva** [comment]
https://github.com/zBritva/charticulator-visual-community/issues/32 related discussion
